### PR TITLE
refactor(Search): improve Chakra implementation

### LINKED ISSUE #9573: DocSearch implementation v2
_After the MVP for DocSearch is merged, iterate it one more time, and apply suggestions given by @TylerAPfledderer in [this comment](https://github.com/ethereum/ethereum-org-website/pull/9509#issuecomment-1442373827)._

### Describe the solution you'd like

_Pasting the entire comment here for readers' convenience. Credits to @TylerAPfledderer_ 

=============================

**tldr**: [CodeSandbox Mock Solution](https://codesandbox.io/s/chakra-w-docsearch-mock-2flf11?file=/src/EthDocSearch/index.tsx)

This ended up being a little bit trickier than I was expecting, but my solution results in having to import `DocSearchButton` and `DocSearchModal` instead of `DocSearch`.

Why? In the `DocSearch` component, the modal is rendered in a portal. If I infuse Chakra props to `DocSearch`, the styling would be scoped to a container I would create, and the modal will be outside this scope. This would mean placing the modal styles as high as the global object in the theme config (not desirable).

## 

So I am using the Chakra Factory to allow chakra props to be available on the components. In addition, because the factory is using emotion to generate a class name hash to attach the props to, I'm ensuring the class name is at least forwarded to the component.

Then, using the `sx` prop to declare the needed className selectors and apply Chakra tokens to the styles where possible.

In the case of a modal, I have to add a wrapper element to pass `className` to because the modal does not directly accept it.

With the button, instead of the factory, because it is using CSS flex, I am consuming it with the Chakra `Flex` component, which I think is more appropriate here in the additional level of abstraction.

I copied all the disclosure logic from `DocSearch` and paste it here. This can now allow for further modification of the logic if you want.

This is open to moving around logic between the components for something that would be more desirable, along with logic that is not here but is in the PR. And the styling might have some room for simplification.

Also, can still consider in the Modal to use it's `hitComponent` and `footerComponent` props to customize the search result styles and footer styles directly with Chakra components instead of through CSS selectors at the higher level.

### Describe alternatives you've considered

After some thought while making this Sandbox, I can not think of a different or better solution here. I will gladly accept any thoughts or observations to explore any changes here. 😄 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
## Description

This PR improves the DocSearch to better utilize Chakra.

- Imports the `DocSearchButton` and `DocSearchModal` with event logic from `DocSearch`
  - Supercharge these components with Chakra to directly accept props.
- Split all the custom styling into two for these components and place the two styling objects in a utils file.

## Nice to have

Would like to add a story for this component but ran into problems related to `react-i18next`. There is a [Storbook addon](https://storybook.js.org/addons/storybook-react-i18next) available to setup the environment, but the dashboard still ran into various errors.

## Related Issue
Closes #9573

### CODE REVIEW

**Review by wackerow — APPROVED** [review]
Thanks @TylerAPfledderer, great refactor! This is working great on my end. Learned a few things here too, appreciate you jumping on this!

**Review by pettinarip — APPROVED** [review]
Nicee job @TylerAPfledderer! loved the usage of the Portal there 👍🏼

File: src/components/Search/utils.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+      color: "bodyLight",
+      textTransform: "uppercase",
+    },
+  },
+
+  "svg[aria-label='Algolia'] *": {
+    fill: "bodyLight",
+  },
+
+  "@media (max-width: 768px)": {
```
[the remark this answers is not in the document]
Comment (TylerAPfledderer) [reply src/components/Search/utils.ts:206]: Doing this
```ts
md: {
  ".DocSearch-Modal": {
    "--inset": "space.4",
    borderRadius: "base",
    inset: "var(--inset)",
    maxWidth: "calc(100vw - 2 * var(--inset))",
    maxHeight: "calc(100vh - 2 * var(--inset))",
    overflow: "hidden",
    position: "fixed",
  },
},
```
will not parse as expected, if this is what you are looking for.

Responsive values are only able to be parsed within the individual style props, which means here that all of those props would get an object value of `{ base: value, md: undefined }` to make this work. =/

I can certainly explore this as an option for the `sx` prop that this whole object is getting sent to, and see if this feature can be added without unexpected side-effects or add too much to the runtime/buildtime.

[the remark this answers is not in the document]
Comment (pettinarip) [reply src/components/Search/utils.ts:200]: No, lets keep it this way then. I was thinking more on the approach of using `useToken` (not sure if it is possible).

```tsx
const mdBp = useToken("breakpoints", "md")

// and then using it like this
[`@media (max-width: ${mdBp})`]: {
  ...
}
```

I don't know if this is even possible xD to have a `rem` value on the media query 🤔

**Review by pettinarip — APPROVED** [review]
🚀

File: src/components/Search/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (TylerAPfledderer) [reply src/components/Search/utils.ts:200]: @pettinarip with how I have it set up, it would mean passing that token value from the hook as an argument.

**Pull request merged**